### ISSUE TITLE
Couldn't connect to [mariadb|postgres] server

### ISSUE THREAD

**Issue by tomhoad** [bitnami/bitnami-docker-express#108]
**Description**

Couldn't connect to mariadb server error on docker-compose up

**Steps to reproduce the issue:**

1. Run `docker-compose up`

**Describe the results you received:**

```
myapp_1    | INFO  ==> Connecting to at mariadb server at 172.22.0.2:3306.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 1.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 2.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 3.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 4.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 5.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 6.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 7.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 8.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | INFO  ==> Trying to connect to mariadb server at 172.22.0.2:3306. Attempt 9.
myapp_1    | /opt/bitnami/express/functions: line 33: nc: command not found
myapp_1    | ERROR ==> Couldn't connect to mariadb server.
```
**Describe the results you expected:**

The application to be available at :3000

**Additional information you deem important (e.g. issue happens only occasionally):**

Occurs with any of the DB docker-compose templates in this repo (maria and postgres).

**Version**

- Output of `docker version`:

```
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:22:38 2018
  OS/Arch:      linux/amd64
  Experimental: true
```

- Output of `docker info`:

```
Containers: 278
 Running: 0
 Paused: 0
 Stopped: 278
Images: 43
Server Version: 18.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.87-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: 6OS2:OOKC:5X5N:TQF5:VGMK:P4WN:YLQE:IQJR:3ENR:2VIR:SMKU:FTCD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 22
 Goroutines: 41
 System Time: 2018-08-25T12:06:23.409381279Z
 EventsListeners: 2
HTTP Proxy: docker.for.mac.http.internal:3128
HTTPS Proxy: docker.for.mac.http.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.21.0, build 5920eb0
docker-py version: 3.2.1
CPython version: 3.6.4
OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Host OS: `Darwin RH-LON-OSXL-55.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 21 20:07:39 PDT 2018; root:xnu-3789.73.14~1/RELEASE_X86_64 x86_64`

**Comment by juan131** [comment]
Hi @tomhoad 

I was able to reproduce the issue! The problem is the image expects **netcat** to be installed. I'm working on fixing the base image so it includes it. I will update this issue as soon as we have more news.

**Comment by juan131** [comment]
Hi @tomhoad 

We just released a new revision of this docker image (**4.16.3-r33**). This new image includes **netcat** and you should be able to run **`docker-compose up`** without any issue. Thanks so much for reporting the bug.

Please feel free to test it and close this issue if you considered it solved!

**Comment by tomhoad** [comment]
Hi @juan131, I can confirm the new images work perfectly. Thanks for this!
